WMF Framework/SharedContainerCache.swift

```diff
@@ -1,20 +1,23 @@
 import Foundation
 
-public final class SharedContainerCache<T: Codable> {
+public final class SharedContainerCache<T: Codable>: SharedContainerCacheHousekeepingProtocol {
     
     // values must all be distinct
     public enum PathComponent: String {
         case widgetCache = "Widget Cache"
         case pushNotificationsCache = "Push Notifications Cache"
         case userDataExportSyncInfo = "User Data Export Sync Info"
+        case talkPageCache = "Talk Page Cache"
     }
     
     private let pathComponent: PathComponent
     private let defaultCache: () -> T
+    private let fileFolder: String
     
-    public init(pathComponent: PathComponent, defaultCache: @escaping () -> T) {
+    public init(pathComponent: PathComponent, fileFolder: String = String(), defaultCache: @escaping () -> T) {
```

Comment:
I realize this is a larger refactor, but I think this class would be much easier to read if we reworked these parameters into:

```
private let fileName: String
private let subdirectoryPathComponent: String?

public init(fileName: String, subdirectoryPathComponent: String?, defaultCache: @escaping () -> T) {
    self.fileName = fileName
    self.subdirectoryPathComponent = subdirectoryPathComponent
    ...
}
```

A class that needs to cache a file in the root directory inits without a subdirectory, like:

```
let sharedCache = SharedContainerCache<PushNotificationsCache>.init(fileName: "Push Notifications Cache", defaultCache: {...})
```

and talk pages can init like:

```
let sharedCache = SharedContainerCache<TalkPageCache>.init(fileName: cachedFileName, subdirectory: "Talk Pages Cache", defaultCache: {
    TalkPageCache(talkPages: [])
})
```

I just think working with the `PathComponent` enum is hurting us more than helping, at this point. Let me know what you think! Most of the rest of my comments will have this refactor in mind.